mod_app/tests/test_support_models.py

```diff
@@ -37,42 +29,70 @@ def test_tag_creation(self):
 class TestLink(TestCase):
     def test_link_creation(self):
         url = models.URLField("some_url.com")
-        link = Link.objects.create(path=url, description="testing link creation")
+        link = Link.objects.create(url=url, description="testing link creation")
 
         self.assertTrue(link)
-        self.assertEqual(link.path, url)
+        self.assertEqual(link.url, url)
         self.assertEqual(link.description, "testing link creation")
-        self.assertIsInstance(link.path, models.URLField)
+        self.assertIsInstance(link.url, models.URLField)
+
+
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+
+class TestFileLinkModels(TestCase):
+    filelink_models = [
+        Script,
+        PressBook,
+        Programme,
+        Publicity,
+        Still,
+        Postcard,
+        Poster,
+        Drawing,
+    ]
+
+    def create_instance(self, model_class):
+        # Create an instance of the specified model with file, url, and description
+        return model_class.objects.create(
+            file=SimpleUploadedFile(
+                "test_file.txt", b"File content goes here.", content_type="text/plain"
+            ),
+            url="https://example.com",
+            description="A test filelink.",
+        )
+
+    def test_filelink_creation(self):
+        for model_class in self.filelink_models:
+            instance = self.create_instance(model_class)
 
+            self.assertTrue(instance)
+            self.assertEqual(instance.url, "https://example.com")
+            self.assertEqual(instance.description, "A test filelink.")
 
-class TestCopy(TestCase):
-    def test_copy_creation(self):
-        copy = Copy.objects.create(name="test copy")
+    def test_upload_to_function(self):
+        for model_class in self.filelink_models:
+            instance = self.create_instance(model_class)
+            expected_path = f"files/{model_class.__name__}/test_file"
```

Comment:
how come this isn't test_file.txt?

Reply:
the `upload_to` function appends the name with a unique string so there are no duplicates so it's like test_file_97fv0.txt